subsys/settings/src/settings_store.c

```diff
@@ -110,6 +109,17 @@ int settings_save(void)
 	}
 	rc = 0;
 
+	Z_STRUCT_SECTION_FOREACH(settings_handler_stat, ch) {
+		if (ch->h_export) {
+			rc2 = ch->h_export(settings_save_one);
+			if (!rc) {
+				rc = rc2;
+			}
+		}
+	}
+
+#if defined(CONFIG_SETTINGS_DYNAMIC_HANDLERS)
+	struct settings_handler *ch;
```

Comment:
You may as well put this block into a helper function e.g setting_save_dynamic to avoid having declarations in the middle of the scope, though the later may just be a convention no really enforced so I fine you decide it doesn't justify writing a new function.

Reply:
I don't think it justifies writing a new function. I have put it there to remove any warning about unused variables if CONFIG_SETTINGS_DYNAMIC_HANDLERS = n. I could of course have added the #if twice but I think it's easier to read this way.